src/hydro/hydro_driver.cpp

```diff
@@ -150,6 +189,48 @@ TaskCollection HydroDriver::MakeTaskCollection(BlockList_t &blocks, int stage) {
         hydro_pkg.get());
   }
 
+  // calculate magnetic tower scaling
+  if ((stage == 1) && hydro_pkg->AllParams().hasKey("magnetic_tower_power_scaling") &&
+      hydro_pkg->Param<bool>("magnetic_tower_power_scaling")) {
+    const auto &magnetic_tower =
+        hydro_pkg->Param<cluster::MagneticTower>("magnetic_tower");
+
+    // need to make sure that there's only one region in order to MPI_reduce to work
+    TaskRegion &single_task_region = tc.AddRegion(1);
+    auto &tl = single_task_region[0];
+    // First globally reset magnetic_tower_linear_contrib and
+    // magnetic_tower_quadratic_contrib
+    auto prev_task =
+        tl.AddTask(none, cluster::MagneticTowerResetPowerContribs, hydro_pkg.get());
+
+    // Adding one task for each partition. Given that they're all in one task list
+    // they'll be executed sequentially. Given that a par_reduce to a host var is
+    // blocking it's also save to store the variable in the Params for now.
+    for (int i = 0; i < num_partitions; i++) {
+      auto &mu0 = pmesh->mesh_data.GetOrAdd("base", i);
+      auto new_magnetic_tower_power_contrib =
+          tl.AddTask(prev_task, cluster::MagneticTowerReducePowerContribs, mu0.get(), tm);
+      prev_task = new_magnetic_tower_power_contrib;
+    }
+#ifdef MPI_PARALLEL
+    auto reduce_magnetic_tower_power_contrib = tl.AddTask(
+        prev_task,
+        [](StateDescriptor *hydro_pkg) {
+          Real magnetic_tower_contribs[] = {
+              hydro_pkg->Param<Real>("magnetic_tower_linear_contrib"),
+              hydro_pkg->Param<Real>("magnetic_tower_quadratic_contrib")};
+          PARTHENON_MPI_CHECK(MPI_Allreduce(MPI_IN_PLACE, magnetic_tower_contribs, 2,
+                                            MPI_PARTHENON_REAL, MPI_MAX, MPI_COMM_WORLD));
+          hydro_pkg->UpdateParam("magnetic_tower_linear_contrib",
+                                 magnetic_tower_contribs[0]);
+          hydro_pkg->UpdateParam("magnetic_tower_quadratic_contrib",
+                                 magnetic_tower_contribs[1]);
+          return TaskStatus::complete;
+        },
+        hydro_pkg.get());
```

Comment:
As above, if this keeps popping up as a pattern we should abstract it (but keep it for now)